src/cache/messages.ml

```diff
@@ -8,10 +8,31 @@ let invalid_args args =
     (Printf.sprintf "invalid arguments:%s"
        (String.concat ~sep:" " (List.map ~f:Sexp.to_string args)))
 
+let version_at_least ~min v =
+  v.major > min.major || (v.major = min.major && v.minor >= min.minor)
+
+let string_of_version { major; minor } = sprintf "%i.%i" major minor
+
+let dyn_of_version { major; minor } =
+  Dyn.Encoder.record
+    [ ("major", Dyn.Encoder.int major); ("minor", Dyn.Encoder.int minor) ]
+
+let hint_min_version = { major = 1; minor = 2 }
```

Comment:
This should probably change to `1.3` since `1.2` is taken by the version without sending full paths in cache.

Reply:
So I thought, but as long as there is no dune public release in-between, we can stash several protocol modifications in one version.

Reply:
Ah, sure, indeed.